### ISSUE TITLE
Rails should seed test databases when it forks them for parallel tests

### ISSUE THREAD

**Issue by Ecco** [rails/rails#40812]
### Steps to reproduce
* Write a test suite that depends on seeds to actually pass.
* See how your tests work well if you `rails db:prepare` then `rails test`, because `db:prepare` will seed your test database
* Enable `parallelize(workers: 2)` and see how your test suite fails

### Expected behavior
The test suite should still pass

### Actual behavior
Rails creates temporary additional test databases (one per process), but doesn't seed them…

### System configuration
**Rails version**: 6.1
**Ruby version**: 2.6.6

**Comment by aar0nr** [comment]
Seed data isn't designed to be used for tests. Use [fixtures](https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html) to set up your test database instead. If you don't like fixtures, you can use [factories](https://github.com/thoughtbot/factory_bot) instead.

**Comment by Ecco** [comment]
I would beg to disagree. Yes, you're right in the general scenario. But what about a "Country" table for example? You probably don't want to hit a Country factory every single time…

Also, rails does seed the database when running tests single-threaded. So there is definitely an issue: either seeds should be considered part of the test states or not part of the state, but that decision shouldn't change if tests are run in parallel or not.

**Comment by aar0nr** [comment]
Regardless if seeds are run, do not rely on them for test data. Use fixtures to insert a few sample countries.

**Comment by Ecco** [comment]
Sure, I guess this would work. But then I think rails shouldn't ever seed any test database 😄

**Comment by estebanbouza** [comment]
Having fixtures could probably be a better technical solution than DB seeds, however the inconsistency of 1 vs 2+ threads is something that needs addressing like @Ecco mentioned.

We are running a 5 year old rails app that always relied on base DB data in order to run tests. Migrating that base DB data back to fixtures would be an immense task and would decrease our confidence in tests. Currently we go from 100% to 40% pass from 1 to 2 threads.

For example our payment plans data, coupons enabling/disabling or account limits are something that goes in data migrations via https://github.com/ilyakatz/data-migrate and allows us to test with a mirror of production data safely.

What would be the "hacky" way of pre-seeding those DBs with parallelism? Are they named as `app_test_0`, `app_test_1` .. for example?

**Comment by Ecco** [comment]
Here's the fix we use @estebanbouza 
```ruby
class ActiveSupport::TestCase
  parallelize_setup do |_worker|
    # For some reason Rails doesn't seed test databases when it forks them
    Rails.application.load_seed
  end
end
```

**Comment by estebanbouza** [comment]
Thanks! Between my last message and this one we ended up doing this:


```ruby
require 'rake'

  parallelize_setup do |worker|
    MyApplication::Application.load_tasks
    Rake::Task['data:migrate'].invoke
  end
  ```
  
  Your code looks much cleaner, I'll try. Thanks!
